12/task1_phonebook.py

```diff
@@ -0,0 +1,140 @@
+"""
+Home Work 11
+Dmytro Verovkin
+robot_dreams
+
+1. Використати файл як базу даних для збереження записів телефонної книги із попередніх завдань.
+
+Ваша телефонна книга, що до цього містилася в dict, має зберігатися у вигляді тексту в JSON форматі.
+При закритті програми і повторному відкритті всі попередні дані мають бути доступними.
+Підказка: Ви можете використати бібліотеку json, яка допоможе із перетворенням dict в JSON string (при записі в файл)
+і JSON string в dict (при читанні із файлу). Файл слід оновлювати після кожної успішної операції add або delete.
+"""
+
+import json
+
+print("PHONE BOOK")
+print("commands: stats, add <name> <phone>, delete <name>, list, show <name>, exit, help")
+help_text = """
+--------------------------------------------------------------------------------------------------
+PHONEBOOK HELP
+
+Use commands to operate this phonebook.json:
+stats              - shows number of phonebook.json entries
+add <name> <phone> - adds <name> and <phone> number to phonebook.json, if <name> is not already exists
+show <name>        - prints name and phone number for entry <name>
+delete <name>      - deletes entry from phonebook.json by <name>
+list               - prints all entries of the phonebook.json
+exit               - exits the program
+help               - shows this help message
+--------------------------------------------------------------------------------------------------
+"""
+
+from os import path
+
+PHONE_BOOK_BD_FILE = 'phonebook.json'
+
+# check if database file is exist. if not - create empty
+if not path.exists(PHONE_BOOK_BD_FILE):
+    with open(PHONE_BOOK_BD_FILE, 'w') as f:
+        empty_phone_book = {}
+        json.dump(empty_phone_book, f)
+
+
+def rewrite_JSONed(file, raw_data):
+    """
+    Converts RAW data into JSON and rewrites to an open file
+    :param file:file instance
+    :param raw_data: data to convert to JSON and save
+    :return:
+    """
+    file.truncate(0)
+    file.seek(0)
+    json.dump(raw_data, file)
+
+
+def get_stat():
+    with open(PHONE_BOOK_BD_FILE, 'r') as f:
+        phone_book = json.load(f)
+        print(f"Total {len(phone_book)} entries")
+
+
+def add_to_phonebook(name, phone):
+    with open(PHONE_BOOK_BD_FILE, 'r+') as f:
```

Comment:
Open the file only in the place where you need it. In your case, this is line 69